processor/processor.go

```diff
@@ -2216,7 +2216,8 @@ func (proc *Handle) transformSrcDest(
 			var successCountMetadataMap map[string]MetricMetadata
 			eventsToTransform, successMetrics, successCountMap, successCountMetadataMap = proc.getDestTransformerEvents(response, commonMetaData, eventsByMessageID, destination, transformer.UserTransformerStage, trackingPlanEnabled, transformationEnabled)
 			failedJobs, failedMetrics, failedCountMap := proc.getFailedEventJobs(response, commonMetaData, eventsByMessageID, transformer.UserTransformerStage, transformationEnabled, trackingPlanEnabled)
-			proc.saveFailedJobs(failedJobs)
+			droppedJobs := proc.getDroppedJobs(response, eventList)
+			proc.saveFailedJobs(append(failedJobs, droppedJobs...))
```

Comment:
I believe that for rudder sources' use cases both failed and dropped jobs in processor's pipeline could be considered as dropped (i.e. non-retryable)

1. We could rename 
  - `proc.saveFailedJobs` to `proc.saveDroppedJobs `
  - `rsources.NewFailedJobsCollector` to `rsources.NewDroppedJobsCollector`
  - `rsources.JobsFailed` to `rsources.JobsDropped`
3. Dropped jobs failed keys shouldn't be captured
4. It would be better to report rsources dropped jobs during `Store`, once for all scenarios as the last operation of the method, after storing router, batchrouter & proc_errors